src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/switch.txt

```diff
@@ -309,3 +309,24 @@ Error({Kind:ErrorKind.InvalidArgument})
 //DateTime-DateTime
 >> Switch("Case1","Case2",DateTimeValue("4/1/2001 10:00:00"),"Case1",DateTimeValue("4/1/2001 10:00:00"))
 DateTime(2001,4,1,10,0,0,0)
+
+// Result of switch can't be consumed.
+>> Abs(Switch(1, 1, -2, {x:4}))
+Errors: Error 0-28: The function 'Abs' has some invalid arguments.|Error 4-27: Invalid argument type (Void). Expecting a Number value instead.
+
+// Expression generating void value can be used inside of the switch. If(1<0, 1, {x:1}) => void value
+>> Switch(1, 1, 2, If(1<0, 1, {x:4}))
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+>> Switch(1, 1, 1/0, 10, {x:4})
+Error({Kind:ErrorKind.Div0})
+
+>> Switch("a", "a", 2, {x:4}, 3)
+Errors: Error 20-25: Invalid argument type (Record). Expecting a Text value instead.|Error 0-29: The function 'Switch' has some invalid arguments.
+
+>> Switch(1, 2, 2, 3, b)
+Errors: Error 19-20: Name isn't valid. 'b' isn't recognized.|Error 0-21: The function 'Switch' has some invalid arguments.
+
+>> Switch(10, 1, 1/0, 10, {x:4})
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
```

Comment:
Can you move this test together with the one on line 321, and add a comment that compares the two? For example: "// Void expressions can become error" in the first, and "// Void expressions that are not errors have a void value"

Reply:
Updated.